packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -75,7 +79,11 @@ public class AuthCognito : FlutterPlugin, MethodCallHandler {
   lateinit var hubEventChannel: EventChannel
   private val authCognitoHubEventStreamHandler: AuthCognitoHubEventStreamHandler
   var eventMessenger: BinaryMessenger? = null
-  private lateinit var token: SubscriptionToken;
+  private lateinit var token: SubscriptionToken
+  private val unknownErrorMessage: String = "An unknown error has occurred when calling the amplify_auth_cognito MethodChannel."
```

Comment:
The error message in the exception and the Exception name should be as developer friendly as possible and it should make sense to the developers (i.e. not leak internal implementation details or rely on developers to understand how amplify-flutter works internally).

We should break these error messages into a generic error message that "An unknown error has occurred" and underlyingException to detail what went wrong in the library to help troubleshooting.

Reply:
Ok cool.  I think I'm going to rename this to 'Unrecognized' error, since unknown isn't exactly accurate.

Reply:
Another reason to change the name to 'Unrecognized'  - Android, at least, actually has an error type called Unknown Exception.  Casting this type to a dart class called 'unknown', while casting unexpected exceptions to 'Unrecognized' could be extremely helpful when debugging or helping with customer issues.